src/core/src/bound_evaluate.cpp

```diff
@@ -470,9 +470,11 @@ bool ov::interval_bound_evaluator(const Node* node,
         if (!lower_output_values[i]) {
             fully_defined = false;
         } else {
-            // Can not set to get_constant_min_of_type(lower_output_values[i]->get_element_type())
-            // yet
-            op::v1::Select().evaluate(lower_out, {final_input_dyn_mask, zero_t, lower_output_values[i]});
+            // Can not set to get_constant_min_of_type(lower_output_values[i]->get_element_type()) yet
+            const auto& lower = lower_output_values[i];
+            const auto typed_zero = op::v0::Constant::create(lower.get_element_type(), Shape{1}, {0});
```

Comment:
why the change is required?

Reply:
Before `zero_t` with element type `int64` was used. That was was fine for `evaluate` method with `HostTensor` parameters, even if the other tensor was of different element type. But with `ov::Tensor` these types need to be identical, otherwise `Tensor::data<T>()` throws.

Reply:
Moreover Select Op specification requires _then_ and _else_ tensors to be of the same type.

Reply:
May it be required to replace `zero_t` with such approach? (for `reduce_max` var etc.)

Reply:
For reduction 2nd input is axes and `zero` Constant is used which got i64.
For evaluate it would fine to give tensor same type or integral to be according spec. But get normalized axes will handle other types also..
I think change for reduction is not required stick to `0` as i64 is ok.